### ISSUE TITLE
[glsl-in] Cannot field select on a lhs value

### ISSUE THREAD

**Issue by JCapucho** [gfx-rs/naga#901]
The following code will produce `Can't lookup field on this type "x"`
```
#version 450
void main() {
    vec4 a = vec4(1.0);
    a.x = 2.0;
}
```
The problem is that `field_selection` will check the type of `a` and this will return a `Pointer` and this isn't handled

**Comment by jakobhellermann** [comment]
If think this isn't completely fixed, it doesn't work when the right hand side is a vector:
```glsl
vec4 color = vec4(1.0);
color.rgb = vec3(2.0);
```
```
[ERROR naga::valid::compose] Vector component[0] type ValuePointer { size: None, kind: Float, width: 4, class: Function }
Function [1] 'main' is invalid:
        Expression [8] is invalid
        Composing 0's component type is not expected
```

**Comment by JCapucho** [comment]
Good catch It seems like like the way we swizzle currently doesn't return us a pointer to a vector when in `lhs` position but always a vector, I'll see if I can fix it

**Comment by Frizi** [comment]
I believe I've fixed this in #973
